User management

MrCondescending opened the pull request:
Finalized user management page and back end logic. Test cases included.

Commit:
initial push to Git

Commit:
A few user changes, pushing for merge up to dev.

Commit:
Merge remote-tracking branch 'remotes/origin/development' into UserManagement

# Conflicts:
#	Riki/user/users.json
#	Riki/wiki/web/templates/base.html

Commit:
Added option to select User Management page when 'admin', created user_manage page (only base), and added forms and routes to accompany them.

Commit:
Commiting so Noah can look and please help me :)

Commit:
Fixed directory issue.

Commit:
Fixed directory issue.

Commit:
omfg

Commit:
Added the rest of the user management options, logic is not complete yet.

Commit:
Finished most of the logic for the application, edit_user is not properly writing to user.json, and test cases still need to be created.

Commit:
Finalized edit_user functionality.

Commit:
Added test cases.

OGoodness reviewed (changes requested):
Need to remove Duckduckgo.com in the html.

The tests fail due to them using a Full directory path from your machine.

Import for forms may need to be changed so that it works for all machines. Had to change to form . import forms

OGoodness committed:
Minor reworks for PR.

Fixed test directory path for user_management.py
Removed unknown duckduckgo.com url
Add try catch for import issues

OGoodness commented:
Issues are now fixed
